lib/manageiq/appliance_console/message_server_configuration.rb

```diff
@@ -0,0 +1,247 @@
+
+    private
+
+    def get_hostname
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      LinuxAdmin::Hosts.new.hostname
+    end
+
+    def configure_log3
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      FileUtils.cp(@log4_properties_sample_path, @log4_properties_path) unless file_found?(@log4_properties_path)
+
+      require 'pry'; binding.pry # JJV
+    end
+
+    def configure_jaas
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      content = <<~EOS
+        KafkaServer {
+          org.apache.kafka.common.security.plain.PlainLoginModule required
+          username="#{@username}"
+          password="#{@password}"
+          user_admin="#{@password}"
+        };
+      EOS
+
+      File.write(@jaas_config_path, content) unless file_found?(@jaas_config_path)
+
+      require 'pry'; binding.pry # JJV
+    end
+
+    def configure_client_properties
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      content = <<~EOS
+        ssl.truststore.location="#{@truststore_path}"
+        ssl.truststore.password="#{@password}"
+
+        sasl.mechanism=PLAIN
+        security.protocol=SASL_SSL
+        sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \\
+          username="#{@username}" \\
+          password="#{@password}";
+      EOS
+
+      File.write(@client_properties_path, content) unless file_found?(@client_properties_path)
+
+      require 'pry'; binding.pry # JJV
+    end
+
+    def configure_ssl
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      content = <<~EOS
+        [ subject_alt_name ]
+      EOS
+
+      File.write(@openssl_config_path, content) unless file_found?(@openssl_config_path)
+
+      require 'pry'; binding.pry # JJV
+    end
+
+    def configure_logs_directory
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      return true if file_found?(LOGS_DIR)
+
+      FileUtils.mkdir_p(LOGS_DIR)
+      FileUtils.chmod(0744, LOGS_DIR)
+      FileUtils.chown("kafka", "kafka", LOGS_DIR)
+
+      require 'pry'; binding.pry # JJV
+    end
+
+    def configure_firewall
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      AwesomeSpawn.run!("firewall-cmd --add-port=9092/tcp --permanent") # insecure
+      AwesomeSpawn.run!("firewall-cmd --add-port=9093/tcp --permanent") # secure
+      AwesomeSpawn.run!("firewall-cmd --reload")
+
+      require 'pry'; binding.pry # JJV
+    end
+
+    def configure_keystore
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      # JJV Use directory: @keystore_path
+
+      require 'pry'; binding.pry # JJV
+    end
+
+    def configure_service_properties
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      # Create the .ORIG only on clean environments
+      FileUtils.cp(@server_properties_path, @server_properties_orig_path) unless file_found?(@server_properties_orig_path)
+
+      content = <<~EOS
+
+        ssl.keystore.location="#{keystore_path}"
+        ssl.keystore.password="#{@password}"
```

Comment:
Do we have to handle escaping or anything in any of these passwords (here and elsewhere we do interpolation)?  Like, what if someone has a `"` in their password?

Reply:
@Fryguy I'll investigate. Thank you!